libraries/apollo-annotations/api/apollo-annotations.api

```diff
@@ -8,6 +8,9 @@ public abstract interface annotation class com/apollographql/apollo/annotations/
 public abstract interface annotation class com/apollographql/apollo/annotations/ApolloInternal : java/lang/annotation/Annotation {
 }
 
+public abstract interface annotation class com/apollographql/apollo/annotations/ApolloPrivateEnumConstructor : java/lang/annotation/Annotation {
+}
+
```

Comment:
This introduces the Opt-in marker that we removed in https://github.com/apollographql/apollo-kotlin/commit/d160e982a8afe8e0e7a58df427cf9f83c4eee2ee.

The underlying reason is that Kotlin doesn't support static factory method (companion obejcts cannot access private constructors of nested classes). 

The previous trick of extracting a public interface doesn't work with sealed classes because the compiler insists on having `UNKNOWN__Impl` be part of the exhaustive `when` statements and it's private in the file. Maybe a future version of the compiler could fix that but as of today I haven't found a way.